### PR TITLE
fix: ignore invalid app_ids in /update and /stats requests

### PR DESCRIPTION
**Pull request by ayewo**
Fix to /claim https://github.com/Cap-go/capgo/issues/384

### CODE REVIEW

**Comment by riderx** [comment]
Hey @ayewo thanks a lot for the fix, my ask was to use zod package to validate the fields, and no only the appId but all fields

**Comment by WcaleNieWolny** [comment]
I do not think you can validate 
all fields. I do not see a way to implement that. Validating only app is is okay as far as I am concerned.

The only things are missing are tests for this. @ayewo Can you please add some tests for this? (See the latest merged PR with backend tests)

This test should test the following edge cases:
 - Empty app name
 - weird app name (like the one Martin shared)
 - a very long app id (that is valid but has 30 dots for example)

File: supabase/functions/stats/index.ts
Comment (WcaleNieWolny) [inline supabase/functions/stats/index.ts]: A platform could a union perhaps?

File: supabase/functions/stats/index.ts
Comment (WcaleNieWolny) [inline supabase/functions/stats/index.ts]: Can we use the default values? Like is_emulator the default would be false and is prod would be by default true

File: supabase/functions/updates/index.ts
Comment (WcaleNieWolny) [inline supabase/functions/updates/index.ts]: I made a mistake while working on this zod schema, if you want you can port the fixed version from [here](https://github.com/Cap-go/capgo/pull/387) or wait until that PR is merged and resolve merge conflicts

File: supabase/functions/updates/index.ts
Comment (WcaleNieWolny) [inline supabase/functions/updates/index.ts]: There is something called a “safeParse” you can use that instead of a try catch block

File: supabase/functions/stats/index.ts
Comment (WcaleNieWolny) [inline supabase/functions/stats/index.ts]: Perhaps use zod safe parse, there should exist a method that does not throw but instead returns the error

**Review by WcaleNieWolny — CHANGES_REQUESTED** [review]
Please resolve the comments I left above, but over all the changes are a big improvement over what we had

File: supabase/functions/stats/index.ts
Comment (ayewo) [reply supabase/functions/stats/index.ts]: When I checked now, nope since it's used as a container for mixing types (e.g string and number).

**Comment by ayewo** [comment]
@WcaleNieWolny thanks for your comments.

I've pulled in the latest from `main` by discarding my commits, but that ended up closing this PR so I opened a fresh PR in #399.

File: supabase/functions/stats/index.ts
Comment (ayewo) [reply supabase/functions/stats/index.ts]: In my testing, `safeParse` is lenient. It will let incorrectly formatted fields through.

it doesn't flag invalid JSON fields unless you wrap `parse` inside a `try / catch`.

File: supabase/functions/updates/index.ts
Comment (ayewo) [reply supabase/functions/updates/index.ts]: See my explanation regarding `safeParse` below.

**Pull request closed without merging**
